pan3d/ui/file_select.py

```diff
@@ -0,0 +1,102 @@
+                    flat=True,
+                    icon="mdi-close",
+                    style="float: right",
+                    click="%s = undefined" % dialog_shown,
+                )
+                vuetify.VCardTitle(
+                    "{{ %s }}" % dialog_message,
+                    v_show=dialog_message,
+                )
+                with html.Div(v_show="!%s" % dialog_message):
+                    vuetify.VCardTitle("{{ %s }} File Select" % dialog_shown)
+
+                    vuetify.VFileInput(
+                        v_model=selected_config_file,
+                        v_show=f"{dialog_shown} === 'Import'",
+                        accept=".json",
+                        label="Config File",
+                    )
+                    vuetify.VBtn(
+                        v_show=f"{dialog_shown} === 'Import' && {selected_config_file}",
+                        variant="tonal",
+                        text=(dialog_shown,),
+                        click=submit_import,
+                        style="width: 100%",
+                    )
+
+                    vuetify.VTextField(
+                        v_model=selected_config_file,
+                        v_show=f"{dialog_shown} === 'Export' && {selected_config_file} != false",
+                        label="Download Location",
+                        prepend_icon="mdi-paperclip",
+                        # FileSystem API is only available on some browsers:
+                        # https://caniuse.com/native-filesystem-api
+                        click="""
+                            if ($event){
+                                try {
+                                    window.showSaveFilePicker({
+                                        suggestedName: 'pan3d_state.json',
+                                        types: [{
+                                            accept: {
+                                                'application/json': ['.json']
+                                            }
+                                        }]
+                                    }).then((handle) => {
+                                        handle.createWritable().then((writable) => {
+                                            writable.write(JSON.stringify(%s, null, 4));
+                                            writable.close();
+                                            %s = 'Export complete.'
+                                        })
+                                    });
+                                } catch {
+                                    %s = false;
+                                    window.alert('Your browser does not support selecting a download location. Your download will be made to the default location, saved as pan3d_state.json.')
+                                }
+                            }
+                        """
+                        % (state_export, dialog_message, selected_config_file),
```

Comment:
f-string

Reply:
Changed in [28561cb](https://github.com/Kitware/pan3d/pull/33/commits/28561cb331878bed9f5131df5a01d8307360c41e)